synapse/logging/opentracing.py

```diff
@@ -909,8 +909,9 @@ def tag_args(func: Callable[P, R]) -> Callable[P, R]:
     @wraps(func)
     def _tag_args_inner(*args: P.args, **kwargs: P.kwargs) -> R:
         argspec = inspect.getfullargspec(func)
+        # We use `[1:]` to skip the `self` object reference
         for i, arg in enumerate(argspec.args[1:]):
-            set_tag("ARG_" + arg, str(args[i]))  # type: ignore[index]
+            set_tag("ARG_" + arg, str(args[i + 1]))  # type: ignore[index]
```

Comment:
Tests possible if you push 😓

Reply:
Maybe more worth doing since it doesn't seem to behave as we assume (https://github.com/matrix-org/synapse/pull/13452#discussion_r937392286)